src/product_representations.jl

```diff
@@ -229,20 +244,18 @@ struct ProductRepr{TM<:Tuple}
 end
 
 ProductRepr(points...) = ProductRepr{typeof(points)}(points)
-eltype(x::ProductRepr) = eltype(Tuple{map(eltype, x.parts)...})
-similar(x::ProductRepr) = ProductRepr(map(similar, x.parts)...)
-similar(x::ProductRepr, ::Type{T}) where T = ProductRepr(map(t -> similar(t, T), x.parts)...)
+eltype(x::ProductRepr) = eltype(Tuple{map(eltype, submanifold_components(x))...})
+similar(x::ProductRepr) = ProductRepr(map(similar, submanifold_components(x))...)
+function similar(x::ProductRepr, ::Type{T}) where {T}
+    return ProductRepr(map(t -> similar(t, T), submanifold_components(x))...)
+end
 
 function copyto!(x::ProductRepr, y::ProductRepr)
-    map(copyto!, x.parts, y.parts)
+    map(copyto!, submanifold_components(x), submanifold_components(y))
     return x
 end
 
-function submanifold_component(x::ProductRepr, i::Integer)
-    return x.parts[i]
-end
-
 (+)(v1::ProductRepr, v2::ProductRepr) = ProductRepr(map(+, v1.parts, v2.parts)...)
```

Comment:
I think you could also change these methods of `+` and `-` to use `submanifold_components` for consistency.